src/Tables.jl

```diff
@@ -565,6 +565,29 @@ struct Partitioner{T}
     x::T
 end
 
+"""
+    Tables.getrows(x, inds; view=nothing)
+
+Return one or more rows from table `x` according to the position(s) specified by `inds`:
+
+- If `inds` is a single integer return a row object.
+- If `inds` is a collection of integers, return an indexable object of rows. 
+  In this case, the returned type is not necessarily the same as the original table type.
```

Comment:
```suggestion
- If `inds` is a single non-boolean integer return a row object.
- If `inds` is a vector of non-boolean integers, a vector of booleans, or a `:`, return an indexable object of rows. 
  In this case, the returned type is not necessarily the same as the original table type.
```

Reply:
Good catch. It would be good to restrict the type signature of methods defined below to disallow undocumented index types. Otherwise, even if we say behavior is undefined some people may start relying on them.

Reply:
either restrict signatures or make a run-time check in the body of the function (I have not checked with would be better - the issue is that restricting signatures is very difficult for indexing). It might be easier to check if the output has a proper shape.